src/window.rs

```diff
@@ -169,6 +184,18 @@ impl<'a, F, W> Iterator for Windower<'a, F, W>
     }
 }
 
+impl<S, W> Iterator for Windowed<S, W>
```

Comment:
This feels suspicious to me. I pretty quickly got stuck in an infinite loop when I just called `collect()` on a `Windowed` signal in some existing code. I think I'd rather have API compilation breakage than silent code failure. Here's an example of my usage:

```rust
let bin = 2048;
for chunk in window::Windower::hanning(&vector[..], bin, 1024) {
    // Why do we need to pass bin again, after we've already told the Window how long it should be?
    let chunk_data: Vec<[f64; 1]> = chunk.take(bin).collect();
}
```